### DISCUSSION TITLE
Varchar always null?

### DISCUSSION THREAD

**Discussion by Phosphorus-M** [mikro-orm/mikro-orm#2892] [Q&A]
I'm not totally sure if this is a bug

Basically, I'm using Mikrorm with Type GraphQL
I'm doing a query for specific field varchar always return null, in all cases

![image](https://user-images.githubusercontent.com/19656993/157838095-9d69e57b-d4da-45e4-8609-279510c5c34b.png)

As I said I'm not sure if this is a bug but it's very similar, I tried with many entities but just fail with this column.
It's not an error or something like that, didn't stop the execution but it's a strange thing. 

![image](https://user-images.githubusercontent.com/19656993/157839127-1ec79a21-0bca-4b4c-8010-62a3dc4bbd5e.png)

It's the configuration of the field, It's varchar of 30 of length, nullable, and is a unique key, nothing more, it's not a special type. 
**To Reproduce**

<details>
<summary>The code</summary>
<div>
This is the class of TagTypes

```TypeScript
import { Entity, PrimaryKey, Property, Unique } from '@mikro-orm/core';
import { Field, ObjectType } from 'type-graphql';

@ObjectType()
@Entity({ tableName: 'FUsTagTypes' })
export class FUsTagTypes {

  @Field()
  @PrimaryKey({fieldName:"tagtype_id", type: "number"})
  tagtypeId!: number;

  @Field({ name: 'name', nullable: true })
  @Property({ nullable: true, default: 'NULL', fieldName:"name" })
  name?: string;

}
```

</div>
</details>

**Additional context**
<details>
<summary>The code</summary>
<div>
Threads call to Tags

```TypeScript
import { Collection, Entity, ManyToMany } from '@mikro-orm/core';
import { Field, ObjectType } from 'type-graphql';
import { FUdTags } from './FUdTags';

@ObjectType("Thread", {description: "Return a list of threads"})
@Entity({ tableName: 'FUdThreads' })
export class Thread {

  @Field(type => [FUdTags], { name: 'tags'})
  @ManyToMany({ entity: () => FUdTags, pivotTable: 'FUrThreads_Tags', joinColumn: 't_id', inverseJoinColumn: 'tag_id' })
  FUrThreadsTags = new Collection<FUdTags>(this);

}

```

Tags call to TagTypes

```TypeScript
import { Collection, Entity, ManyToMany, ManyToOne } from '@mikro-orm/core';
import { Thread } from './Thread';
import { FUsTagTypes } from './FUsTagTypes';
import { Field, ObjectType } from 'type-graphql';

@ObjectType()
@Entity({ tableName: 'FUdTags' })
export class FUdTags {

  @Field(() => FUsTagTypes, { nullable: true })
  @ManyToOne({ entity: () => FUsTagTypes, fieldName: 'tagtype_id', nullable: true, defaultRaw: `0`, index: 'fk_tag_tagtype' })
  tagtype?: FUsTagTypes;

  @Field(() => [Thread])
  @ManyToMany({ entity: () => Thread, pivotTable: 'FUrThreads_Tags', joinColumn: 'tag_id', inverseJoinColumn: 't_id' })
  FUrThreadsTags = new Collection<Thread>(this);

}

```

</div>
</details>

**Versions**

| Dependency | Version |
| - | - |
| node | 17.6.0 |
| typescript | ^4.5.5 |
| @mikro-orm/core | ^5.0.5 |
| @mikro-orm/mysql | ^5.0.5 |
| type-graphql | ^1.1.1 |
| graphql | ^15.7.2 |

**Comment by B4nan** [comment]
`default: 'NULL'` means literally a default value which is a string "NULL". Either use `default: null` (without quotes) or `defaultRaw: 'null'`.

If you need more help, provide reproduction _without_ GQL involved, can't help with that.

**Comment by B4nan** [comment]
Wait a sec, this looks like you are defining the pivot entity explicitly, while still using m:n. This is not supported, you either define just the m:n and let the pivot entity be generated, or you define all 3 entities, but don't use m:n - as in this example https://mikro-orm.io/docs/composite-keys#use-case-3-join-table-with-metadata
